deepmd_utils/model_format/output_def.py

```diff
@@ -0,0 +1,278 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    List,
+    Tuple,
+    Union,
+)
+
+
+def check_var(var, var_def):
+    if var_def.atomic:
+        # var.shape == [nf, nloc, *var_def.shape]
+        if len(var.shape) != len(var_def.shape) + 2:
+            raise ValueError(f"{var.shape[2:]} length not matching def {var_def.shape}")
+        if list(var.shape[2:]) != var_def.shape:
+            raise ValueError(f"{var.shape[2:]} not matching def {var_def.shape}")
+    else:
+        # var.shape == [nf, *var_def.shape]
+        if len(var.shape) != len(var_def.shape) + 1:
+            raise ValueError(f"{var.shape[1:]} length not matching def {var_def.shape}")
+        if list(var.shape[1:]) != var_def.shape:
+            raise ValueError(f"{var.shape[1:]} not matching def {var_def.shape}")
```

Comment:
**Suggestion:** Consider using a more descriptive name than `var` and `var_def` in the `check_var` function. Good variable names make the code more readable and maintainable. [maintainability]
```suggestion
def check_var(variable, variable_definition):
    if variable_definition.atomic:
        # variable.shape == [nf, nloc, *variable_definition.shape]
        if len(variable.shape) != len(variable_definition.shape) + 2:
            raise ValueError(f"{variable.shape[2:]} length not matching def {variable_definition.shape}")
        if list(variable.shape[2:]) != variable_definition.shape:
            raise ValueError(f"{variable.shape[2:]} not matching def {variable_definition.shape}")
    else:
        # variable.shape == [nf, *variable_definition.shape]
        if len(variable.shape) != len(variable_definition.shape) + 1:
            raise ValueError(f"{variable.shape[1:]} length not matching def {variable_definition.shape}")
        if list(variable.shape[1:]) != variable_definition.shape:
            raise ValueError(f"{variable.shape[1:]} not matching def {variable_definition.shape}")
```